hal/src/thumbv6m/sercom/v2/uart/pad.rs

```diff
@@ -0,0 +1,452 @@
+        }
+    }
+
+    /// Set the `CTS` [`Pad`]
+    #[inline]
+    pub fn cts<I>(self, pin: impl AnyPin<Id = I>) -> Pads<S, RX, TX, RTS, Pad<S, I>>
+    where
+        I: GetPad<S>,
+        Pad<S, I>: IsPad,
+    {
+        Pads {
+            sercom: self.sercom,
+            receive: self.receive,
+            transmit: self.transmit,
+            ready_to_send: self.ready_to_send,
+            clear_to_send: pin.into().into_mode(),
+        }
+    }
+}
+
+/// Define a set of [`Pads`] using [`PinId`]s instead of [`Pin`]s
+///
+/// In some cases, it is more convenient to specify a set of `Pads` using
+/// `PinId`s rather than `Pin`s. This alias makes it easier to do so.
+///
+/// The first type parameter is the [`Sercom`], while the remaining four are
+/// effectively [`OptionalPinId`]s representing the corresponding type
+/// parameters of [`Pads`], i.e. `RX`, `TX`, `RTS` & `CTS`. Each of the
+/// remaining type parameters defaults to [`NoneT`].
+///
+/// ```
+/// use atsamd_hal::pac::Peripherals;
+/// use atsamd_hal::gpio::v2::{PA08, PA09, Pins};
+/// use atsamd_hal::sercom::v2::{Sercom0, uart};
+/// use atsamd_hal::typelevel::NoneT;
+///
+/// pub type Pads = uart::PadsFromIds<Sercom0, PA08, PA09>;
+///
+/// pub fn create_pads() -> Pads {
+///     let peripherals = Peripherals::take().unwrap();
+///     let pins = Pins::new(peripherals.PORT);
+///     uart::Pads::default().rx(pins.pa09).tx(pins.pa08)
+/// }
+/// ```
+///
+/// [`Pin`]: crate::gpio::v2::Pin
+/// [`PinId`]: crate::gpio::v2::PinId
+/// [`OptionalPinId`]: crate::gpio::v2::OptionalPinId
+
+#[cfg(feature = "samd21")]
+pub type PadsFromIds<S, RX = NoneT, TX = NoneT, RTS = NoneT, CTS = NoneT> = Pads<
+    S,
+    <RX as GetOptionalPad<S>>::Pad,
+    <TX as GetOptionalPad<S>>::Pad,
+    <RTS as GetOptionalPad<S>>::Pad,
+    <CTS as GetOptionalPad<S>>::Pad,
+>;
+
+//=============================================================================
+// PadSet
+//=============================================================================
+
+/// Type-level function to recover the [`OptionalPad`] types from a generic set
+/// of [`Pads`]
+///
+/// This trait is used as an interface between the [`Pads`] type and other
+/// types in this module. It acts as a [type-level function], returning the
+/// corresponding [`Sercom`] and [`OptionalPad`] types. It serves to cut down on
+/// the total number of type parameters needed in the [`Config`] struct. The
+/// [`Config`] struct doesn't need access to the [`Pin`]s directly.  Rather, it
+/// only needs to apply the [`SomePad`] trait bound when a `Pin` is required.
+/// The [`PadSet`] trait allows each [`Config`] struct to store an instance of
+/// [`Pads`] without itself being generic over all six type parameters of the
+/// [`Pads`] type.
+///
+/// [`Pin`]: crate::gpio::v2::Pin
+/// [type-level function]: crate::typelevel#type-level-functions
+/// [`Config`]: super::Config
+pub trait PadSet: Sealed {
```

Comment:
A while back, I renamed `spi::AnyPads` to `spi::PadSet`. At the time, I think the type-level mapping was a bit more complicated, which made it more complicated to implement the full `AnyKind` pattern. However, with the latest approach, `PadSet` is basically `AnyPads` without the `AsRef` and `AsMut` implementations.

There's really no use case for *most* of the `AnyKind` machinery, like `SpecifcPads` or the super traits. *But*, defining this as `AnyPads` might make it clearer that it follows the exact same pattern for the same purposes.

What do you think? Should we convert this back to `AnyPads` or stick with `PadSet`?

Reply:
I like `PadSet`. To me it makes it clearer that it's "a set of pads for the specified SERCOM" rather than just "any set of Pads". Plus, I think the `Any_` naming convention should probably be reserved specifically for `AnyKind` implementations.

Reply:
I wasn't proposing to keep the trait the same yet call it `AnyPads`. I was proposing to implement the full `AnyKind` pattern for `AnyPads`. It's just that most features would go unused. You wouldn't ever need `SpecificPads` nor any of the super trait bounds. However, the one advantage would be consistency. I'm not sure if that's worth it though.